src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.ml;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.ml.client.MachineLearningNodeClient;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.MLInputDataset;
+import org.opensearch.ml.common.dataset.TextDocsInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+import org.opensearch.ml.common.model.MLModelTaskType;
+import org.opensearch.ml.common.output.model.ModelResultFilter;
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.output.model.ModelTensors;
+
+/**
+ * This class will act as an abstraction on the MLCommons client for accessing the ML Capabilities
+ */
+@RequiredArgsConstructor
+@Log4j2
+public class MLCommonsClientAccessor {
```

Comment:
Can we call this MLInferenceService instead of MLCommonsClientAccessor?

Reply:
As the MLClient is not an Inference Service, it provides more functions apart from inference, like load model etc. 

Moreover, XXXXClientAccessor is a standard pattern and it is just another way to abstract the functionalities of MLClient.

Reply:
I bet to disagree on this. The search service and cluster service are not using any client to build their functionality. Hence calling MLservice here won't be correct. 

We generally write accessors on top of our clients.

Reply:
Another name i can think of MLCommonsAccessor.

Reply:
> I bet to disagree on this. The search service and cluster service are not using any client to build their functionality. Hence calling MLservice here won't be correct.

No, I dont think it is necessarily incorrect. From OpenSearch, services typically provide an interface in front of one component of the system. In addition, they have a lifecycle and are sometimes hierarchical. Generally, they abstract away implementation specific details.

If we made this into a service, we would abstract all interactions with ml-commons into this service (and potentially subservices). In addition to wrapping client calls, we could add additional features such as statistic tracking and additional validation logic.

For now, we can keep it as is, but in future, when we want to add additional functionality, we may want to change to a service.

Reply:
This just another class and not a service and neither a lifecycle component, as it is not maintaining any lifecycle.

> If we made this into a service, we would abstract all interactions with ml-commons into this service (and potentially subservices). In addition to wrapping client calls, we could add additional features such as statistic tracking and additional validation logic.

The abstraction of all the MLCommons functionality is already abstracted via MLClient. Other features like validation should be done at layer above this and not in this class as it will bloat the functionality of this class. The primary purpose this class is to provide abstraction for MLClient so that if any changes happen in the MLClient interfaces those details are not penetrated to above layers.


> No, I dont think it is necessarily incorrect. From OpenSearch, services typically provide an interface in front of one component of the system. In addition, they have a lifecycle and are sometimes hierarchical. Generally, they abstract away implementation specific details.

The implementation details are already abstracted away from Neural plugin the moment we had MLClient with us. Now we also want abstract the functionality of MLClient hence we created this. 

We can think of this similar to ModelDao in k-NN plugin over there we were calling ClusterService or other services as that was the only interface to call those service, or lets say we want to call DDB of AWS we can put a DDBAccesor around it which can call DDBClient, but won't create a Service for that as the lifecycle part is already abstracted via DDBClient.